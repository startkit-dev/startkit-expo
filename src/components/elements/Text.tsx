import { clsx } from "clsx"
import {
  Text as DefaultText,
  type TextProps as DefaultTextProps
} from "react-native"

export type TextProps = DefaultTextProps & {
  /**
   * Toggle to use the default monospace font
   */
  mono?: boolean
}

/**
 * A Wrapper round the <Text> component which applies our basic styles,
 * considering dark mode, etc.
 */
export function Text({ mono, ...props }: TextProps) {
  return (
    <DefaultText
      className={clsx(
        "text-black dark:text-white",
        mono && "font-mono font-normal"
      )}
      {...props}
    />
  )
}
