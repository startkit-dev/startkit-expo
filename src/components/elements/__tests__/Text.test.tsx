import { create } from "react-test-renderer"

import { Text } from "../Text"

test("renders correctly", () => {
  const tree = create(<Text>Text test</Text>).toJSON()

  expect(tree).toMatchSnapshot()
})

test("renders mono text", () => {
  const tree = create(<Text mono>Mono text test</Text>).toJSON()

  expect(tree).toMatchSnapshot()
})
