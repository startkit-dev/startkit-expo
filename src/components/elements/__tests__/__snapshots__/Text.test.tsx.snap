// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text>
  Text test
</Text>
`;

exports[`renders mono text 1`] = `
<Text>
  Mono text test
</Text>
`;
